# This comes from upstream, it's the non buggy part of 
# https://code.qt.io/cgit/qt/qtbase.git/commit/?id=9c56d4da2ff631a8c1c30475bd792f6c86bda53c
# https://code.qt.io/cgit/qt/qtbase.git/commit/?id=813a928c7c3cf98670b6043149880ed5c955efb9
# the buggy parts were later reverted
--- a/src/corelib/global/qendian.h	2022-01-14 21:38:15.785002200 +0100
+++ b/src/corelib/global/qendian.h	2022-01-14 21:38:25.834038200 +0100
@@ -44,6 +44,8 @@
 #include <QtCore/qfloat16.h>
 #include <QtCore/qglobal.h>
 
+#include <limits>
+
 // include stdlib.h and hope that it defines __GLIBC__ for glibc-based systems
 #include <stdlib.h>
 #include <string.h>
--- a/src/corelib/global/qfloat16.h	2022-01-14 21:38:46.834101400 +0100
+++ b/src/corelib/global/qfloat16.h	2022-01-14 21:38:40.276265400 +0100
@@ -43,6 +43,7 @@
 
 #include <QtCore/qglobal.h>
 #include <QtCore/qmetatype.h>
+#include <limits>
 #include <string.h>
 
 #if defined(QT_COMPILER_SUPPORTS_F16C) && defined(__AVX2__) && !defined(__F16C__)
--- a/src/corelib/text/qbytearraymatcher.h	2022-01-14 21:39:14.134141000 +0100
+++ b/src/corelib/text/qbytearraymatcher.h	2022-01-14 21:39:08.111327200 +0100
@@ -42,6 +42,8 @@
 
 #include <QtCore/qbytearray.h>
 
+#include <limits>
+
 QT_BEGIN_NAMESPACE
 
 
--- a/src/3rdparty/angle/src/libANGLE/HandleAllocator.cpp	2022-01-14 21:51:02.562375800 +0100
+++ b/src/3rdparty/angle/src/libANGLE/HandleAllocator.cpp	2022-01-14 21:50:54.522118600 +0100
@@ -10,6 +10,7 @@
 #include "libANGLE/HandleAllocator.h"
 
 #include <algorithm>
+#include <limits>
 
 #include "common/debug.h"

